File: harmonize/harmonize/defs/job.py
import uuid
from datetime import datetime
from enum import Enum
from typing import TypedDict


class Status(Enum):
    SUCCEEDED = 0
    RUNNING = 1
    FAILED = 2

class Job(TypedDict):
    id: uuid.UUID
    started_on: datetime
    description: str
    status: Status

File: app/Harmonize/Harmonize.Client/Model/Job/Job.cs
using System.Text.Json.Serialization;

namespace Harmonize.Client.Model.Job;

public class Job
{
    public required Guid Id { get; set; }
    public required DateTime StartedOn { get; set; }
    public required string Description { get; set; }
    public required string? ErrorMessage { get; set; }
    public required JobStatus Status { get; set; }
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum JobStatus
{
    Succeeded = 0,
    Running = 1,
    Failed = 2,
    Canceled = 3,
}
